stainless_extraction/src/krate.rs

```diff
@@ -328,27 +327,38 @@ impl<'l, 'tcx> BaseExtractor<'l, 'tcx> {
     let mut flags = carrier_flags.to_stainless(f);
 
     // Add flag specifying that this function is a method of its class (if there's a class)
-    flags.extend(class_def.into_iter().map(|cd| {
-      let flag: st::Flag<'l> = f.IsMethodOf(cd.id).into();
-      flag
-    }));
+    flags.extend(
+      class_def
+        .iter()
+        .map(|cd| -> st::Flag<'l> { f.IsMethodOf(cd.id).into() }),
+    );
 
     // Extract the function itself
-    type Parts<'l> = (Params<'l>, st::Type<'l>, st::Expr<'l>);
-    let Generics { tparams, txtcx, .. } = self.get_generics(fn_item.def_id);
+    let Generics {
+      tparams,
+      txtcx,
+      trait_bounds,
+    } = self.get_generics(fn_item.def_id);
+
+    // If this function is not on a class *but* has trait bounds, we need to add
+    // these as evidence parameters.
+    let ev_params = if class_def.is_none() {
+      self.evidence_params(trait_bounds)
+    } else {
+      vec![]
+    };
 
-    let (params, return_tpe, body_expr): Parts<'l> =
-      self.enter_body(hir_id, txtcx.clone(), class_def, |bxtor| {
+    let (params, return_tpe, body_expr): (Params<'l>, st::Type<'l>, st::Expr<'l>) = self
```

Comment:
I kinda liked the type alias here, but if you think it's more readable you can leave it as is.

Reply:
As we're using the three things in as individual parameters later, I prefer this which makes it clear, that they're just grouped for convenience.